Use backticks to render URLs etc as monospaced (#159)
AMWA-TV/nmos-discovery-registration: docs/5.1. Data Model - Identifier Mapping.md

A new Source is created in the following circumstances:

- At the input of essence into a networked media enviroment from the 'edge' (ie. a legacy or incompatible protocol)

> - At the point of capture (ie. in a network connected camera or microphone)
> - At the mix point of two or more Flows of essence (ie. an audio mix bus, vision mix bus)

Reviewer: ```suggestion
- At the point of capture (i.e. in a network connected camera or microphone)
- At the mix point of two or more Flows of essence (i.e. an audio mix bus, vision mix bus)
```

- Where a multiplexed Source is split into multiple mono-essence Sources
- Where a set of Flows are combined as a single multiplexed Source

> Whenever a new Source is generated, its immediate parent Source(s) MUST be recorded. This is achieved via a `parents` attribute in the Node API Sources resource. This contains an array of Source IDs corresponding to the Sources whose Flows have been brought together at this point. For example, a vision mixer which is fading between a Flow from Source A, and a Flow from Source B would indicate ["A", "B"\] in its `parents` attribute for the duration of this transition.

Reviewer: I see a removed backslash before the open square bracket, but not before the close square bracket, is that right? But not backticking this? Hmm, equiv array below is, so suggestion:

```suggestion
Whenever a new Source is generated, its immediate parent Source(s) MUST be recorded. This is achieved via a `parents` attribute in the Node API Sources resource. This contains an array of Source IDs corresponding to the Sources whose Flows have been brought together at this point. For example, a vision mixer which is fading between a Flow from Source A, and a Flow from Source B would indicate `["<id of A>", "<id of B>"]` in its `parents` attribute for the duration of this transition.
```

Author: Strange, I remember changing that one already.

### Flows

A new Flow is created in the following circumstances:

- Where a Flow passes through a processing element which is modelled as a new Source

> - Where a Flow is modified in a non-editorial manner (ie. audio volume changed, EQ applied, video contrast changed, video encoded or decoded)

Reviewer: ```suggestion
- Where a Flow is modified in a non-editorial manner (i.e. audio volume changed, EQ applied, video contrast changed, video encoded or decoded)
```

Note that in order to avoid creating a new Flow after every single processing element held within a Device, processing blocks can be logically grouped together provided it will never be possible to use a Flow from part way through the processing block. For example, a three band EQ block within an audio mixer will cause a new Flow to be generated. Creating a new Flow after each EQ band is an option, but as these Flows could never be used directly anywhere else internal or external to the Device the representation of it is of no benefit. It is however important to register the EQ process as a whole as a new Flow as it might be possible to route the pre-EQ and post-EQ essence to a Sender, or to a mixing element within the Device.

Sources and Flows MUST be advertised via a Node's API, and registered via the Registration API if available. This SHOULD only be done on the Device which originates the Source or Flow; if a Flow is received by a Device from the network it does not need to be re-advertised via APIs.

> Whenever a new Flow is generated, its immediate parent Flow(s) MUST be recorded. This is achieved via a `parents` attribute in the Node API Flows resource. This contains an array of Flow IDs corresponding to the Flows which have been brought together at this point. For example, an encoder which codes a raw video Flow C into a coded video Flow D would indicate `["C"]` in its `parents` attribute.

Reviewer: ```suggestion
Whenever a new Flow is generated, its immediate parent Flow(s) MUST be recorded. This is achieved via a `parents` attribute in the Node API Flows resource. This contains an array of Flow IDs corresponding to the Flows which have been brought together at this point. For example, an encoder which codes a raw video Flow C into a coded video Flow D would indicate `["<id of C>"]` in its `parents` attribute.
```

SHOULD change if:

- Device ID changes
- A different physical interface (such as an SDI input) is used to retrieve essence
- Format URN changes between `video`, `audio`, `data` and `mux` variants

SHOULD NOT change if:

- Configuration parameters associated with the Source are changed, such as its operating resolution or bitrate
- Labels, descriptions or other metadata associated with the Source resource are modified

It is suggested that Source IDs be generated using a combination of:

- The parent Device ID
- A processing element type and index (or name) within the Device
- A processing element Source index (or name)

### Flow ID

Owned by:

- Source ID

SHOULD Change if:

- Source ID changes
- Format URN changes between `video`, `audio`, `data` and `mux` variants (including between two codec types)

SHOULD NOT change if:

- Configuration parameters associated with the Flow are changed, such as its operating resolution or bitrate
- Labels, descriptions or other metadata associated with the Flow resource are modified

It is suggested that Flow IDs be generated using the combination of:

- The parent Source ID
- A processing element type and index (or name) within the Device

> - A processing element Flow index (or name)\

Reviewer: ```suggestion
- A processing element Flow index (or name)
```

System

- Fixed Node ID
- Fixed Device ID

Input

- Fixed Receiver ID

Output

> - Flow ID generated based on input Flow ID (see *Source and Flow
>     Representation* above)

Reviewer: Just consistent with below
```suggestion
- Flow ID generated based on input Flow ID (see *Source and Flow Representation* above)
```
